modules/ensemble/lib/action/get_network_info_action.dart

```diff
@@ -0,0 +1,135 @@
+import 'package:ensemble/framework/action.dart';
+import 'package:ensemble/framework/device.dart';
+import 'package:ensemble/framework/event.dart';
+import 'package:ensemble/framework/scope.dart';
+import 'package:ensemble/framework/stub/location_manager.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:get_it/get_it.dart';
+
+import '../framework/stub/network_info.dart';
+
+class GetNetworkInfoAction extends EnsembleAction {
+  NetworkInfoManager networkInfo = GetIt.I<NetworkInfoManager>();
+  EnsembleAction? onSuccess;
+  EnsembleAction? onError;
+  EnsembleAction? onDenied;
+  EnsembleAction? onLocationDisabled;
+
+  GetNetworkInfoAction({super.initiator,
+    super.inputs,
+    this.onSuccess,
+    this.onError,
+    this.onDenied,
+    this.onLocationDisabled});
+
+  factory GetNetworkInfoAction.from({Invokable? initiator,dynamic payload}) =>
+      GetNetworkInfoAction.fromYaml(initiator: initiator, payload: Utils.getYamlMap(payload));
+  factory GetNetworkInfoAction.fromYaml({Invokable? initiator, Map? payload}) {
+    return GetNetworkInfoAction(
+        initiator: initiator,
+        inputs: Utils.getMap(payload?['inputs']),
+        onSuccess: EnsembleAction.from(payload?['onSuccess'], initiator: initiator),
+        onError: EnsembleAction.from(payload?['onError'], initiator: initiator),
+        onDenied: EnsembleAction.from(payload?['onDenied'], initiator: initiator),
+        onLocationDisabled: EnsembleAction.from(payload?['onLocationDisabled'], initiator: initiator));
+  }
+
+  @override
+  Future<dynamic> execute(BuildContext context, ScopeManager scopeManager) {
+    networkInfo.getLocationStatus().then((locationStatus) {
+      if ((locationStatus == LocationPermissionStatus.denied.name
+          || locationStatus == LocationPermissionStatus.deniedForever.name)
+          && onDenied != null) {
+        return ScreenController().executeAction(context, onDenied!,
+            event: EnsembleEvent(initiator,
+                data: {'status': locationStatus}));
+      } else if (locationStatus == LocationStatus.disabled.name
+          && onLocationDisabled != null) {
+        return ScreenController().executeAction(context, onLocationDisabled!,
+            event: EnsembleEvent(initiator,
+                data: {'status': locationStatus}));
+      } else
+      if (locationStatus == LocationPermissionStatus.unableToDetermine.name
+          && onError != null) {
+        return ScreenController().executeAction(context, onError!,
+            event: EnsembleEvent(initiator,
+                data: {
+                  'status': locationStatus
+                }));
+      } else {
+        try {
```

Comment:
You want to throw error if onSuccess is not specified? If you want to force it required, throw a LanguageError if not specified instead.

Reply:
will do, thanks